### PR TITLE
AWS S3 Upload Routes

### LINKED ISSUE #46: AWS S3 Upload Routes
Allow for uploading of images. We may need to be able to upload multiple images.

- Create a route for image upload
- save to amazon s3 and get the url

If using Node.js, consider using https://www.npmjs.com/package/multer

### PR DESCRIPTION
**Pull request by jethoo**
### What this PR does (required):
Post request to : /upload/images , uploads images to aws S3 bucket as well as writes to the submission database and also returns array of urls from s3 for that particular upload.

### Screenshots / Videos (required):
Screen Shot 2021-07-14 at 23 21 33
Screen Shot 2021-07-14 at 23 23 35
Screen Shot 2021-07-14 at 23 23 49

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/controllers/upload.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+				}
+                const body = req.body
+                //saving into the database
+				let submission = Submission.create({
+						contest: body.contestId,
+						user: body.userId,//req.user.id,
+						active: body.active,
+						storeUrl: galleryImgLocationArray
+				})
+
```
Comment (rajivtitus) [inline server/controllers/upload.js]: Please fix spacing/indentation issues :)

File: server/controllers/upload.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+				let fileArray = req.files,
+				fileLocation;
+				const galleryImgLocationArray = [];
+				for ( let i = 0; i < fileArray.length; i++ ) {
+					fileLocation = fileArray[ i ].location;
+					galleryImgLocationArray.push( fileLocation )
+				}
+                const body = req.body
+                //saving into the database
+				let submission = Submission.create({
```
Comment (rajivtitus) [inline server/controllers/upload.js]: Please add error handling logic, incase we run into problems while creating a document.

File: server/controllers/upload.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	const extname = filetypes.test( path.extname( file.originalname ).toLowerCase());
+	const mimetype = filetypes.test( file.mimetype );
+	if( mimetype && extname ){
+		return cb( null, true );
+	} else {
+		cb( 'Error: Images Only!' );
+	}
+}
+
+const uploadsContestGallery = multer({
```
Comment (rajivtitus) [inline server/controllers/upload.js:25]: 👍🏽

**Review by rajivtitus — APPROVED** [review]
Good work :)

Just a couple of comments, please have a look

File: server/controllers/upload.js (hunk 1 above)
Comment (jeewanbhusal-1) [reply server/controllers/upload.js]: sure :)

File: server/controllers/upload.js (hunk 3 above)
Comment (jeewanbhusal-1) [reply server/controllers/upload.js:25]: @rajivtitus I removed submission object creation logic. Therefore, post request to /uploads/images would not create submission object and will only return urls of the images from s3.

Comment (rajivtitus) [reply server/controllers/upload.js:25]: Sounds great! You can merge this branch and please convey the same information to your team. Thanks!

Comment (jeewanbhusal-1) [reply server/controllers/upload.js:25]: Thank You. I already shared the s3 credentials in the slack group and also the info.

**Pull request merged**